extension/src/popup/helpers/useSetupSigningFlow.ts

```diff
@@ -0,0 +1,138 @@
+import { useEffect, useRef, useState } from "react";
+import { useSelector } from "react-redux";
+import { AppDispatch } from "popup/App";
+import { signTransaction, rejectTransaction } from "popup/ducks/access";
+
+import { Account } from "@shared/api/types";
+
+import {
+  allAccountsSelector,
+  confirmPassword,
+  hardwareWalletTypeSelector,
+  hasPrivateKeySelector,
+  makeAccountActive,
+  publicKeySelector,
+} from "popup/ducks/accountServices";
+
+import {
+  ShowOverlayStatus,
+  startHwSign,
+  transactionSubmissionSelector,
+} from "popup/ducks/transactionSubmission";
+
+export function useSetupSigningFlow(
+  dispatch: AppDispatch,
```

Comment:
do you need to pass dispatch? or can you just instantiate a new one with `useDispatch`? Might make the API a little nicer to have one fewer param

Reply:
yeah 100%, moved in 06295ca863c4e3674866541069298a58a539e699